### PR TITLE
Fix errors/warnings when NUMA enabled

### PR DESCRIPTION
**Pull request by kab163**
Discovered and fixed a couple bugs and warnings when NUMA is enabled. Fixed the ruby CI job that is supposed to test a numa enabled run so that we catch these issues sooner in the future.

### CODE REVIEW

File: examples/cookbook/recipe_move_between_numa.cpp [hunk 1]
```diff
@@ -26,7 +27,7 @@ int main(int, char**)
   auto host_nodes = umpire::numa::get_host_nodes();
 
   if (host_nodes.size() < 1) {
-    UMPIRE_ERROR("No NUMA nodes detected");
+    UMPIRE_ERROR(umpire::runtime_error, umpire::fmt::format("No NUMA nodes detected: {}", host_nodes.size()));
```
Comment (mcfadden8) [inline examples/cookbook/recipe_move_between_numa.cpp]:
```suggestion
    UMPIRE_ERROR(umpire::runtime_error, "No NUMA nodes detected");
```

File: examples/cookbook/recipe_move_between_numa.cpp [hunk 2]
```diff
@@ -46,15 +47,15 @@ int main(int, char**)
 
     // The pointer shouldn't change even though the memory location changes
     if (dst_ptr != src_ptr) {
-      UMPIRE_ERROR("Pointers should match");
+      UMPIRE_ERROR(umpire::runtime_error, umpire::fmt::format("Pointers should match: {}, {}", dst_ptr, src_ptr));
```
Comment (mcfadden8) [inline examples/cookbook/recipe_move_between_numa.cpp]:
```suggestion
      UMPIRE_ERROR(umpire::runtime_error, umpire::fmt::format("Pointers should match: dst_ptr: {} != src_ptr: {}", dst_ptr, src_ptr));
```

File: examples/cookbook/recipe_move_between_numa.cpp [hunk 3]
```diff
@@ -74,7 +75,7 @@ int main(int, char**)
...
```
Comment (mcfadden8) [inline examples/cookbook/recipe_move_between_numa.cpp]:
```suggestion
      UMPIRE_ERROR(umpire::runtime_error, umpire::fmt::format("Pointers should match: dst_ptr: {} != src_ptr: {}", dst_ptr, src_ptr));
```

File: examples/cookbook/recipe_move_between_numa.cpp (hunk 1 above)
Comment (kab163) [reply examples/cookbook/recipe_move_between_numa.cpp]: Can I redeem myself by saying I was multitasking when doing this? Lol your suggestion seems much more reasonable here than what I was about to do!

**Comment by kab163** [comment]
Thanks for the suggested edits, Marty!

**Review by davidbeckingsale — APPROVED** [review]

**Pull request merged**